src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -0,0 +1,774 @@
+        private static SqlChangeOperation GetChangeOperation(IReadOnlyDictionary<string, string> row)
+        {
+            string operation = row["SYS_CHANGE_OPERATION"];
+            switch (operation)
+            {
+                case "I": return SqlChangeOperation.Insert;
+                case "U": return SqlChangeOperation.Update;
+                case "D": return SqlChangeOperation.Delete;
+                default: throw new InvalidDataException($"Invalid change type encountered in change table row: {row}");
+            };
+        }
+
+        /// <summary>
+        /// Builds the command to update the global state table in the case of a new minimum valid version number.
+        /// Sets the LastSyncVersion for this _userTable to be the new minimum valid version number.
+        /// </summary>
+        /// <param name="connection">The connection to add to the returned SqlCommand</param>
+        /// <param name="transaction">The transaction to add to the returned SqlCommand</param>
+        /// <returns>The SqlCommand populated with the query and appropriate parameters</returns>
+        private SqlCommand BuildUpdateTablesPreInvocation(SqlConnection connection, SqlTransaction transaction)
+        {
+            string updateTablesPreInvocationQuery = $@"
+                DECLARE @min_valid_version bigint;
+                SET @min_valid_version = CHANGE_TRACKING_MIN_VALID_VERSION({this._userTableId});
+
+                DECLARE @last_sync_version bigint;
+                SELECT @last_sync_version = LastSyncVersion
+                FROM {SqlTriggerConstants.GlobalStateTableName}
+                WHERE UserFunctionID = '{this._userFunctionId}' AND UserTableID = {this._userTableId};
+                
+                IF @last_sync_version < @min_valid_version
+                    UPDATE {SqlTriggerConstants.GlobalStateTableName}
+                    SET LastSyncVersion = @min_valid_version
+                    WHERE UserFunctionID = '{this._userFunctionId}' AND UserTableID = {this._userTableId};
+            ";
+
+            return new SqlCommand(updateTablesPreInvocationQuery, connection, transaction);
+        }
+
+        /// <summary>
+        /// Builds the query to check for changes on the user's table (<see cref="RunChangeConsumptionLoopAsync()"/>).
+        /// </summary>
+        /// <param name="connection">The connection to add to the returned SqlCommand</param>
+        /// <param name="transaction">The transaction to add to the returned SqlCommand</param>
+        /// <returns>The SqlCommand populated with the query and appropriate parameters</returns>
+        private SqlCommand BuildGetChangesCommand(SqlConnection connection, SqlTransaction transaction)
+        {
+            string selectList = string.Join(", ", this._userTableColumns.Select(col => this._primaryKeyColumns.Contains(col) ? $"c.{col}" : $"u.{col}"));
```

Comment:
If a column was deleted from the user table, would this query fail? In that case would the user have to restart the function in order for userTableColumns to be updated?

Reply:
Thanks for point out. Dropping non-primary-key columns should result in query failure Restarting the user application should re-populate the column list and fix the error. I do not know of a way to resolve this problem as querying table schema every time would be too expensive.

Reply:
We should probably call out in the docs that any modification to the table schema will result in undefined behavior for any triggers currently watching for changes - that'll cover any such scenarios.

Reply:
Done in Readme file under 'Known Issues' section.